frontend/src/main/scala/bloop/data/SettingsMapper.scala

```diff
@@ -0,0 +1,106 @@
+package bloop.data
+import bloop.engine.BuildLoader
+import bloop.logging.Logger
+import bloop.logging.DebugFilter
+import java.nio.charset.StandardCharsets
+import bloop.config.ConfigEncoderDecoders
+import io.circe.Printer
+import io.circe.Json
+import io.circe.parser
+import bloop.io.AbsolutePath
+import java.nio.file.Files
+import bloop.config.Config.Settings
+import io.circe.JsonObject
+import _root_.bloop.config.Config
+import bloop.DependencyResolution
+import scala.util.Try
+
+object SettingsMapper {
+
+  /**
+    * @param settingsPath Path to the json settings file.
+    * @param logger The logger that collects messages about loading the settings.
+    * @return Settings if the file exists.
+    */
+  def fromFile(settingsPath: AbsolutePath, logger: Logger): Option[Settings] = {
+    if (settingsPath.isFile) {
+      val bytes = Files.readAllBytes(settingsPath.underlying)
+      logger.debug(s"Loading workspace settings from ${BuildLoader.settingsFileName}")(
+        DebugFilter.All
+      )
+      val contents = new String(bytes, StandardCharsets.UTF_8)
+      parser.parse(contents) match {
+        case Left(failure) => throw failure
+        case Right(json) => Some(fromJson(json))
+      }
+    } else {
+      None
+    }
+  }
+
+    /**
+    * @param json Json object to extract settings from.
+    * @return Settings extracted from the json.
+    */
+  def fromJson(json: Json): Settings = {
+    ConfigEncoderDecoders.settingsDecoder.decodeJson(json) match {
+      case Right(settings) => settings
+      case Left(failure) => throw failure
+    }
+  }
+
+    /**
+    * @param target Path to where the json settings file should be saved.
+    * @param settings The settings to save.
+    */
+  def save(target: AbsolutePath, settings: Settings): Unit = {
+    val jsonObject = ConfigEncoderDecoders.settingsEncoder(settings)
+    val output = Printer.spaces4.copy(dropNullValues = true).pretty(jsonObject)
+    Files.write(target.underlying, output.getBytes(StandardCharsets.UTF_8))
+  }
+
+    /**
+    * @param settings SemanticDB settings to use.
+    * @param project PRoject to add smanticDB plugin to.
+    * @param logger The logger that collects messages about resolving dependencies.
+    * @return Project with additional scala settings for semanticDB plugin.
+    */
+  def transformProject(
+      settings: Config.Settings,
+      project: Project,
+      logger: Logger
+  ): Project = {
+
+    def findSemanticDBPlugin(scalaVersion: String) =
+      Try {
+        DependencyResolution
```

Comment:
@jvican Do you mean we should use ZincComponentManager directly? And I think this is in conflict to the `latest.release`, since we can't vache that reliably.

Reply:
> Do you mean we should use ZincComponentManager directly?

We should probably use our own component manager only for semanticdb plugins. It'll be very similar to ZincComponentnManager.
  
> And I think this is in conflict to the latest.release, since we can't cache that reliably.

Why not?

Reply:
We don't know what is `latest.release` unless we resolve it . The other way is to never cache `latest.release`, so I guess it should be fine.